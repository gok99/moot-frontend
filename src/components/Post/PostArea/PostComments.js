import React, { useEffect, useState } from 'react';
import { Row } from 'react-bootstrap';

import CommentBox from './CommentBox';

import '../../Styles/styles.css';
import '../post.css';

const PostComments = (props) => {
  const fb = props.fb;
  const comments = props.comments;
  const uid = props.uid;
  const postUid = props.postUid;
  const posterUid = props.posterUid;
  const friends = props.friends;
  const [friendsUidList, setFriendsUidList] = useState([]);
  const commentedPostKey = props.commentedPostKey;
  
  useEffect(() => {
    setFriendsUidList(friends.map((friend) => friend.uid));
    console.log(friends);
    console.log(friendsUidList);
  }, []);

  if (comments.length === 0) {
    return (
      <Row>
        <p className="d-flex justify-content-start text-post content">This post has no comments.</p>
      </Row>
    );
  } else {
    const commentsList = comments.map((commentObj) => {
      const posterIdentity = friendsUidList.includes(commentObj.uid) ? friendsUidList.filter((uid) => uid === commentObj.uid).username : "anonymous user";
      // console.log(friendsUidList);
      // console.log(friendsUidList.filter((uid) => uid === commentObj.uid));
      // console.log(commentObj.uid);
      // console.log(0);
      const poster = 
        commentObj.uid === uid
          ? "me"
          : commentObj.uid === posterUid
            ? "OP (" + posterIdentity + ")"
            : posterIdentity;

      return <Row><CommentBox commentKey={commentObj.key} commentContent={commentObj.comment} commentTime={commentObj.commentTime} myComment={commentObj.uid === uid} fb={fb} uid={uid} postUid={postUid} posterUid={posterUid} commentedPostKey={commentedPostKey} poster={poster}></CommentBox></Row>
    });
    return (
      <Row>{commentsList}</Row>
    );
  }
  
}

export default PostComments;