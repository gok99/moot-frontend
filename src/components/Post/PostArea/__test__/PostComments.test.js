import React from 'react';
import ReactDOM from 'react-dom';
import PostComments from '../PostComments';

import { cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer';

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement('div');
  ReactDOM.render(<PostComments uid={0} postUid={0} posterUid={0} commentedPostKey={0} friends={[]} comments={[]} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("matches snapshot", () => {
  const tree = renderer.create(<PostComments uid={0} postUid={0} posterUid={0} commentedPostKey={0} friends={[]} comments={[]} />).toJSON();
  expect(tree).toMatchSnapshot();
});
