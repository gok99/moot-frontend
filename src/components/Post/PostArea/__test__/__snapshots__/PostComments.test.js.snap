// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="row"
>
  <p
    className="d-flex justify-content-start text-post content"
  >
    This post has no comments.
  </p>
</div>
`;
